src/core/agent/services/identifierService.ts

```diff
@@ -208,6 +208,10 @@ class IdentifierService extends AgentService {
       isDeleted: true,
     });
 
+    await this.props.signifyClient.identifiers().update(identifier, {
+      name: `${metadata.theme}-${randomSalt()}:${metadata.displayName}`,
```

Comment:
The theme should be `XX`, this is our indication that it's been deleted, not just the random salt. The random salt is only there so that if we try to delete identifiers with the same name over and over again they won't conflict.